src/History/HistoryListener.php

```diff
@@ -77,7 +78,11 @@ public function handleSuccess(WorkerMessageHandledEvent $event): void
 
         $event->addStamps($stamp->markFinished());
 
-        $this->storage->save($event->getEnvelope(), $this->createResults($event->getEnvelope()));
+        $this->storage->save(
+            $event->getEnvelope(),
+            $this->serializer->encode($event->getEnvelope()->withoutAll(MonitorStamp::class)),
```

Comment:
Should we maybe serialize `$event->getEnvelope()->getMessage()`? I am thinking about the rerun option. It may be good to keep the stamps so that they are also dispatched. Maybe not really needed? WDYT?

Reply:
Envelope will have two MonitorStamp one added by `HistoryListener::addMonitorStamp` (that has only properties **runId** and **dispatchedAt** initialized) this will fail the encoding because method **HistoryListener::transport** will throw an exception whne property transport is not set, to keep them will mean to modify also the MonitorStamp class to have initial null properties.